vcd/resource_vcd_nsxt_edgegateway_static_route.go

```diff
@@ -0,0 +1,342 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdNsxtEdgeGatewayStaticRoute() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdNsxtEdgeGatewayStaticRouteCreate,
+		ReadContext:   resourceVcdNsxtEdgeGatewayStaticRouteRead,
+		UpdateContext: resourceVcdNsxtEdgeGatewayStaticRouteUpdate,
+		DeleteContext: resourceVcdNsxtEdgeGatewayStaticRouteDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtEdgeGatewayStaticRouteImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"edge_gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID for Static Route configuration",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of Static Route",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of Static Route",
+			},
+			"network_cidr": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Network CIDR (e.g. 192.168.1.1/24) for Static Route",
+			},
+			"next_hop": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				MinItems:    1,
+				Description: "A set of next hops to use within the static route",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ip_address": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "IP Address of next hop",
+						},
+						"admin_distance": {
+							Type:        schema.TypeInt,
+							Required:    true,
+							Description: "Admin distance of next hop",
+						},
```

Comment:
Can we do validation in sets?
```suggestion
						"admin_distance": {
							Type:        schema.TypeInt,
							Required:    true,
							Description: "Admin distance of next hop",
							ValidateFunc: IsIntAndAtLeast(1)
						},
```

Reply:
`IsIntAndAtLeast` would expect `TypeString` (tailored for our custom need to support `0`. This one doesn't require it so I used this one `validation.IntAtLeast(1)`

Ran tests.